### PR TITLE
Add ArcadeDB PropertyGraphStore integration

### PR DESCRIPTION
**Pull request by lvca**
## Description

New partner integration package: `llama-index-graph-stores-arcadedb`

[ArcadeDB](https://arcadedb.com) is a multi-model database with native Bolt protocol and OpenCypher support (97.8% TCK compliant). This integration implements `PropertyGraphStore` using the standard `neo4j` Python driver — no APOC procedures required.

### Motivation

ArcadeDB fills the gap left by the [archived Kuzu integration](https://github.com/run-llama/llama_index/issues/17820) as a lightweight, embeddable graph database option. It combines graph, document, key-value, time-series, and vector storage in a single engine.

### Key design decisions

- **Single vertex type**: ArcadeDB enforces one type per vertex, so semantic labels (`PERSON`, `CITY`) are stored as a `label` property on the `Entity` vertex type, with `Chunk` as a separate type for text chunks.
- **Pure Cypher**: All APOC procedures are replaced with standard Cypher — `MERGE` for upserts, variable-length `MATCH (n)-[*1..depth]-()` for `get_rel_map`, split outgoing/incoming queries for `get_triplets`.
- **Vector search**: Native HNSW via ArcadeDB SQL (`SELECT ... WHERE embedding NEAR [...]`) with Python cosine-similarity fallback.
- **Type creation on init**: `CREATE VERTEX TYPE Entity/Chunk IF NOT EXISTS` + unique indexes.

### Features

| Feature | Supported |
|---|---|
| Structured (Cypher) queries | ✅ |
| Vector similarity search | ✅ (native HNSW + fallback) |
| Schema introspection | ✅ (sample-based) |
| Bolt protocol | ✅ (native) |

## New Package

- [x] I have completed the `[tool.llamahub]` section of `pyproject.toml`
- [x] I have provided a detailed `README.md`

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Testing

- **24 unit tests** with mocked `neo4j` driver — covers all `PropertyGraphStore` methods, helper functions, edge cases
- **15 integration tests** — skip automatically when `ARCADEDB_BOLT_URL` is not set
- All unit tests pass locally

## How to test locally

```bash
cd llama-index-integrations/graph_stores/llama-index-graph-stores-arcadedb
pip install -e .
pytest tests/test_pg_stores_arcadedb_unit.py -v
```

For integration tests with a running ArcadeDB instance:

```bash
docker run -d --name arcadedb \
  -p 2480:2480 -p 7687:7687 \
  -e JAVA_OPTS="-Darcadedb.server.plugins=Bolt:com.arcadedb.bolt.BoltProtocolPlugin" \
  arcadedata/arcadedb:latest

curl -X POST http://localhost:2480/api/v1/server \
  -d '{"command":"CREATE DATABASE testdb"}' -u root:playwithdata

ARCADEDB_BOLT_URL=bolt://localhost:7687 pytest tests/test_pg_stores_arcadedb.py -v
```

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: llama-index-integrations/graph_stores/llama-index-graph-stores-arcadedb/llama_index/graph_stores/arcadedb/arcadedb_property_graph.py [hunk 1]
```diff
@@ -0,0 +1,789 @@
...
+            self._url,
+            auth=(self._username, self._password),
+            **config,
+        )
+
+        try:
+            self._driver.verify_connectivity()
+        except Exception as e:
+            msg = f"Could not connect to ArcadeDB at {self._url}: {e}"
+            raise ConnectionError(msg) from e
```
Comment (AstraBert) [inline llama-index-integrations/graph_stores/llama-index-graph-stores-arcadedb/llama_index/graph_stores/arcadedb/arcadedb_property_graph.py:177]: Connectivity checks should not be done within the constructor as they require network I/O but constructors cannot be asynchronous

File: llama-index-integrations/graph_stores/llama-index-graph-stores-arcadedb/llama_index/graph_stores/arcadedb/arcadedb_property_graph.py [hunk 2]
```diff
@@ -0,0 +1,789 @@
...
+
+        self._ensure_types()
```
Comment (AstraBert) [inline llama-index-integrations/graph_stores/llama-index-graph-stores-arcadedb/llama_index/graph_stores/arcadedb/arcadedb_property_graph.py:181]: Also this executes statements, best to avoid in a constructor

**Comment by lvca** [comment]
@AstraBert is it good now?

**Comment by AstraBert** [comment]
All tests are failing, I am assuming it is a problem with them not being up-to-date with the current implementation?

```text
          raise AssertionError(msg)
E           AssertionError: Expected 'verify_connectivity' to have been called once. Called 0 times.
```

Also, linting is failing too, to make that work, run:

```bash
uv pip install pre-commit
pre-commit install
pre-commit run -a
```

From the root folder of the LlamaIndex repo

**Comment by lvca** [comment]
Used claude to fix it, seems to go. Please let me know

File: llama-index-integrations/graph_stores/llama-index-graph-stores-arcadedb/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+Repository = "https://github.com/run-llama/llama_index"
+
+[dependency-groups]
+test = [
+    "pytest>=7.3.0,<8.0.0",
+    "pytest-mock>=3.10.0,<4.0.0",
+]
+dev = [
+    "llama-index-core",
+]
```
Comment (AstraBert) [inline llama-index-integrations/graph_stores/llama-index-graph-stores-arcadedb/pyproject.toml]: This seems to be causing some troubles in the tests: the fact that is reported within the required dependencies should be enough, no need to duplicate within the dev group

File: llama-index-integrations/graph_stores/llama-index-graph-stores-arcadedb/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    "ANN204",
+    "ANN401",
+    "TRY201",
+    "ARG002",
+    "N803",
+    "TC002",
+    "TC003",
+    "TRY300",
+    "N806",
+]
```
Comment (AstraBert) [inline llama-index-integrations/graph_stores/llama-index-graph-stores-arcadedb/pyproject.toml:69]: Not the biggest fan of ignores here and of the noqas in the code: unless there are structural needs that the linting tools catch as errors, I would iterate over the feedback from `pre-commit run -a` to fix errors. Most of the errors can be generally auto-fixed by ruff

**Comment by lvca** [comment]
Ok tried to address both points from your latest comments.

**Comment by logan-markewich** [comment]
Sorry, right now we are pausing contributions that contribute net-new packages. I appreciate the contribution but going to close this out.

**Comment by lvca** [comment]
Thanks @logan-markewich, understood about the freeze on net-new packages.

In the meantime, users can install the ArcadeDB integration as a standalone package from PyPI:

```
pip install llama-index-graph-stores-arcadedb
```

Maintained by @stevereiner at https://github.com/stevereiner/arcadedb-llama-index.

We'll be happy to resubmit for monorepo inclusion when contributions reopen.
